Sam bjs2 68712

SAM19991 opened the pull request.

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
Task BJS2-68712

SAM19991 committed:
bjs268712

SAM19991 committed:
Make gradlew executable

SAM19991 committed:
bjs268712

miked4king commented on src/main/java/school/faang/bjs268712/SpellEvent.java:
```diff
@@ -0,0 +1,32 @@
+    private int id;
+    private String eventType;
+    private String action;
+    private static int counter = 0;
```
статические переменные общепринято объявлять первыми в классе, давай поправим

miked4king commented on src/main/java/school/faang/bjs268712/SpellEvent.java:
```diff
@@ -0,0 +1,32 @@
+
+    public SpellEvent(String eventType, String action) {
+        this.id = ++counter;
+        if (eventType == null) {
+            log.error("EventType cannot be null. " +
+                    "Please provide a valid EventType instance to initialize the SpellEvent.");
+            throw new IllegalArgumentException("EventType cannot be null. A valid EventType instance is required.");
+        }
```
круто что добавил валидацию, но давай пока не будем на ней фокусироваться, т.к мы самостоятельно передаем значения в конструктор, дальше по курсу будем проходить валидацию)
предлагаю пока не делать валидацию аргументов чтобы не раздувать код

miked4king commented on src/main/java/school/faang/bjs268712/HogwartsSpells.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    public void deleteSpellEvent(int id) {
+        if (!spellById.containsKey(id)) {
+            log.error("Invalid spell id: {}", id);
+            throw new IllegalArgumentException("Invalid spell id: " + id);
+        }
```
в строке 37 ты пытаешься удалить, и если такой элемент был, то он будет сохранен в переменной eventToRemove (кстати над названием переменной можно подумать)
так может мы проверим на null данную переменную и если она не null то продолжим удаление в другой коллекции?

SAM19991 committed:
Moved static variables to the top
Removed  validation
In the delete method renamed the variable (to removedEvent)
Added a null-check to delete method

SAM19991 committed:
Moved static variables to the top
Removed  validation
In the delete method renamed the variable (to removedEvent)
Added a null-check to delete method

SAM19991 committed:
Moved static variables to the top
Removed  validation
In the delete method renamed the variable (to removedEvent)
Added a null-check to delete method
